src/Compilers/CSharp/Portable/Symbols/NativeIntegerTypeSymbol.cs

```diff
@@ -25,8 +25,9 @@ internal sealed class NativeIntegerTypeSymbol : WrappedNamedTypeSymbol
         internal NativeIntegerTypeSymbol(NamedTypeSymbol underlyingType) : base(underlyingType, tupleData: null)
         {
             Debug.Assert(underlyingType.TupleData is null);
-            Debug.Assert(!underlyingType.IsNativeIntegerType);
+            Debug.Assert(!underlyingType.IsNativeIntegerTypeOrNumericIntPtr);
```

Comment:
consider adjusting the name or doc comment on the `NativeIntegerTypeSymbol` to indicate this type is only used when the *numeric IntPtr* feature is not available.